.github/workflows/main.yml

```diff
@@ -209,7 +209,8 @@ jobs:
         cargo doc --no-deps --workspace \
           --exclude wasmtime-cli \
           --exclude test-programs \
-          --exclude cranelift-codegen-meta
+          --exclude cranelift-codegen-meta \
+          --features winch
```

Comment:
@jameysharp (and possibly @alexcrichton) for additional context, adding this feature here allows the docs creation to pass. Attempting to add it the the `wasmtime/Cargo.toml` doesn't for some reason 🤔 (at least when run locally)

Reply:
Ah yeah the only thing that I see that could possibly explain it is [this](https://github.com/bytecodealliance/wasmtime/blob/main/crates/wasmtime/Cargo.toml#L15). The curious part is adding `winch` to this doesn't fix the error when I run the `cargo doc` command locally 🤔

Reply:
@jameysharp I spent some time on this issue today, but still can't quite figure out why `wasmtime-environ` is expecting a `component_compiler` implementation even though the feature is not explicitly enabled for `cargo doc` and as far as I can tell, it's not set as default feature anywhere in the workspace. One thing that I discovered though, is that everything works as expected if the `component-model` feature is passed to the `cargo doc` invocation in the github workflow, which would match the invocation that doc.rs will make according to docs metadata entry:

```toml
[package.metadata.docs.rs]
rustdoc-args = ["--cfg", "nightlydoc"]
features = ["component-model"]
```
So I think that using the `component-model` feature in the CI workflow is preferred over using `winch` since this is what docs.rs will use for the documentation anyway. 

I was wondering if you'd be good with this change as a short term fix and I'll circle back with Alex when is back.

Reply:
Your previous comment around the doc attribute was added in https://github.com/bytecodealliance/wasmtime/pull/6119/commits/4d16393b202f594c79eca7ea080a4a9bf65e6fc6 and the approach described in my previous comment was added in https://github.com/bytecodealliance/wasmtime/pull/6119/commits/82e119485ed68dba122655139e9c6d2393aa6b64.

Reply:
Yup, that feels entirely reasonable to me. I appreciate that you also added comments to the CI config mentioning the docs.rs configuration. Would you also add the reverse direction—a brief comment in the docs.rs metadata pointing to the CI config? With that, I think this is good to merge.